app/models/concerns/hostext/search.rb

```diff
@@ -171,6 +171,19 @@ def search_by_config_group(key, operator, value)
         {:conditions => opts, :include => :hostgroup}
       end
 
+      def cast_facts(key, operator, value)
+        is_int = (value =~ /\A[-+]?\d+\z/ ) or (value.is_a?(Integer))
+        is_pg = ActiveRecord::Base.connection.instance_of? ActiveRecord::ConnectionAdapters::PostgreSQLAdapter
+        if (is_int and !is_pg)
```

Comment:
nitpick: afair a while ago we decided against using logical operators based on their name and swap to symobs, e.g.  
```ruby
is_int && !is_pg
```